### ISSUE TITLE
Understanding S.C. 2.3.3 Animation from Interaction - clarify who is responsible for providing the feature to disable?

### ISSUE THREAD

**Issue by PICARDBRB** [w3c/wcag#2250]
Our organization is adopting some Level AAA criterion including 2.3.3.  Is the intent of the criteria for the web content author to provide the feature to disable the animation?

Ref, C39: Using the CSS reduce-motion query to prevent motion relies on the user settings for reduce motion.  If the user has not set reduce motion, the query is useless for that user and website.  Motion will not be disabled on the web content.

If the intent is to honor user preferences only, then C39 is sufficient, however if the intent of the criteria is to provide a mechanism to disable animation then content authors would need a 2nd technique to supplement C39 for all users.

Can you review and provide clarifications both in WCAG and here.  

Thanks for your assistance.

**Comment by patrickhlauke** [comment]
As with most criteria, if a mechanism/functionality is already provided by user agents, it doesn't need to be replicated. I suppose the SC could/should have been prefaced with "A mechanism is available..."

As C39 is indicated as a sufficient technique, then yes - reacting correctly to the user's preference that they prefer reduced motion (provided that setting is available/exposed on all platforms you're targetting) is a viable way to pass the criterion. (noting also that this isn't limited to CSS animations ... you can query the state/result of that media feature via JS and then do other things like stopping JS-based animations, videos, etc - evaluate `window.matchMedia('(prefers-reduced-motion: reduce)')` and use that to toggle things on/off via JS too)

**Comment by PICARDBRB** [comment]
**But as ever what counts is the end result for the user** - Ditto.  With interpretation we look at the end user result.....so, if not exposed we need a 2nd method...in the wild we cannot guarantee that a mechanism is available on all platforms, can we?  What are your thoughts in updating sufficient techniques....I understand in other SC there are references to a combination of techniques to meet the criterion.  Do you think it valuable to incorporate in this S.C.?

**Comment by patrickhlauke** [comment]
this comes down to the (admittedly handwavy in WCAG) concept of "accessibility supported" https://w3c.github.io/wcag/21/guidelines/#dfn-accessibility-supported

we can't guarantee anything across all possible user agents old and new. but you have to make a pragmatic call about the platforms you intend to support, other platforms that are out there, and whether or not it's a reasonable assumption that users that do require a particular feature have tools/user agents readily available that support those features they need

**Comment by alastc** [comment]
> If the intent is to honor user preferences only, then C39 is sufficient, however if the intent of the criteria is to provide a mechanism to disable animation then content authors would need a 2nd technique to supplement C39 for all users.

As Patrick mentioned, C39 is listed as "sufficient", therefore you don't have to provide another method.

There is a certain amount of flexibility (/ambiguity) because technology moves faster than guidelines, and the situation can be different in different areas of the world. E.g. if most of your population is on old android devices that (hypothetically) don't support reduced motion, then the answer might be different.

By design WCAG aims _not_ to embed technical solutions into the core guidance.

The [techniques section of the understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions#techniques) seems clear, would something else have helped understand that C39 was sufficient?

**Comment by PICARDBRB** [comment]
Level AAA is more than the minimum and C39 implies the minimum.  There is a lot of reading between the lines on this one.  In our experience we have seen literal interpretations without context and judgement..._it's a sufficient technique, so it's sufficient and will meet WCAG_.  

What would help is wording similar to 2.2.2 pause stop hide - "A mechanism is available.." in the definition and within C39 and Applicability section beneath CSS statement: "Users that set motion preferences".

Thoughts?

**Comment by patrickhlauke** [comment]
> In our experience we have seen literal interpretations without context and judgement...*it's a sufficient technique, so it's sufficient and will meet WCAG*.

but that is exactly what "sufficient technique" means. the working group has deemed that technique to be sufficient in meeting the SC.

the whole discussion about accessibility supported technologies etc. is fundamental to most/all of WCAG itself, so not really something that should be replicated on a per-SC basis.

> What would help is wording similar to 2.2.2 pause stop hide - "A mechanism is available.." in the definition

that would require a change to the normative SC text - this sort of change can take quite a while to be agreed/passed, just as a word of warning.

>  and within C39 and Applicability section beneath CSS statement: "Users that set motion preferences"

to be more pedantic, it applies to users that browser with a user agent/operating system combination that allows them to set motion preferences (i.e. they have the capability to do it, not necessarily that they *have* done it explicitly already)

**Comment by PICARDBRB** [comment]
Agreed with sufficient technique, where the feature is supported.....context implied.  Where we have an audience that uses Internet Explorer, the feature is not supported and therefore using the reduced motion media query would be insufficient to meet the success criteria.

Appreciate the discussion on this and additional information.  I would like to see C39 updated for applicability as you stated.  I think our issues are related directly to the technique versus the normative text which would be a nice to have.  

Applicability:   "Applies to users with a user agent/operating system combination that allows them to set motion preferences."

Your clarification on if it has been done already is helpful to our internal discussions.

Thanks.

**Comment by alastc** [comment]
The applicability is supposed to be for the technology rather than user-agents, e.g. Applicable to CSS and JavaScript. 

I think you'd struggle to find systems where it doesn't apply now, it has been part of iOS and Android for around 10 years, and Windows & Mac not to far after that, with the default browsers + Chrome (I think) supporting it.

Internet Explorer has pretty much disappeared from any [stats](https://gs.statcounter.com/browser-market-share) I've seen for a while, even the UK public sector has moved on! (Parts held out on IE for a loooonnnggg time.)

**Comment by PICARDBRB** [comment]
If we look at end of life support to determine audiences versus various stats reported by various organizations:

- 06/15/2022: EOL for Internet Explorer (note Edge can be used on Windows 8.1)
- 01/10/2023: EOL for Windows 8.1
- 12/05/2017: EOL Android Oreo (old where Android Pie (v9) supports the setting)

Using C39 would provide coverage for the majority of audiences after June this year, and given the Success Criteria for this discussion is Level AAA which may not be widely used, there is little risk of missing some audiences using older platforms given vendor EOL support.

C39 relies on users' awareness of the platform setting to reduce motion.  The technique describes using the query but mentions the reliance on enabling reduce motion in the "tests" section.  Would it be valuable to update the description _"The objective of this technique makes use of the 'prefers-reduced-motion' CSS Media Query by honoring the user setting for the combination of user agent/operating system preventing animation from being displayed on Web pages"__?
